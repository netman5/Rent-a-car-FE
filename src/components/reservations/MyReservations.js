/* eslint-disable react/jsx-key */
/* eslint-disable no-tabs */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { getMyResevationsFromApi } from '../../redux/reducers/myReservations';
import styles from './MyReservations.module.css';

const MyReservations = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const currentReservations = useSelector((state) => state.reservation);

  useEffect(() => {
    dispatch(getMyResevationsFromApi());
  }, []);

  const handleClick = ((e) => {
    navigate(`/cars/${e.target.id}`);
  });
  return (
    <>
      <ul>
        {
          (currentReservations.length === 0)
            ? (
              <div className={styles.noReservations}>
                <p className="text-center">No cars reserved</p>
              </div>
            )
            : (
              <table className="table-success">
                <thead>
                  <tr>
                    <th scope="col">Pick up date</th>
                    <th scope="col">Drop off date</th>
                    <th scope="col">Pick up city</th>
                    <th scope="col">Drop off city</th>
                    <th scope="col">Car details</th>
                  </tr>
                </thead>
                <tbody>
                  {currentReservations.map((item) => (
                    <tr>
                      <th scope="row">{item.id}</th>
                      <td>{item.pick_up_date}</td>
                      <td>{item.drop_off_date}</td>
                      <td>{item.pick_up_city}</td>
                      <td>{item.return_city}</td>
                      <td><button type="button" id={item.car_id} onClick={((e) => { handleClick(e); })}>Car details</button></td>
                    </tr>
                  ))}
                </tbody>
              </table>
            )
			}
      </ul>
    </>
  );
};

export default MyReservations;
