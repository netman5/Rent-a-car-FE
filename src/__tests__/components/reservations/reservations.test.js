import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import MyReservations from '../../../components/reservations/MyReservations';
import store from '../../../redux/configureStore';

it('MyReservations renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <MyReservations />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
