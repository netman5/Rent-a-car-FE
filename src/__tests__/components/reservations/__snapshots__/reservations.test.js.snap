// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservations renders correctly 1`] = `
<ul>
  <div
    className="noReservations"
  >
    <p
      className="text-center"
    >
      No cars reserved
    </p>
  </div>
</ul>
`;
